lib/type-checker.js

```diff
@@ -398,10 +393,20 @@ class TypeChecker {
   }
 
   _stringTypes(string, highPrecisionSupport) {
+    let types = [ 'String', 'Object', 'Array' ];
     var passing = find(highPrecisionSupport ? HP_STRING_TESTS : STRING_TESTS, (test) => {
       return test.tester.test(string);
     });
-    return passing ? passing.types : [ 'String', 'Object', 'Array' ];
+    if (passing) {
+      types = passing.types;
+    }
+    // Add date dynamically to the existing types lists, as it is on a
+    // completely different dimension to numeric types and so significantly
+    // more complicated to enumerate all valid combinations
+    if (DATE_CHECK.test(string)) {
+      types = ['Date', ...types];
```

Comment:
and maybe also change the array spread operator to a simple types.unshift('Date') to remain backwards compatible.